geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/ManageBackupBucketMessage.java

```diff
@@ -128,36 +132,53 @@ public static NodeResponse send(InternalDistributedMember recipient, Partitioned
    * indefinitely for the acknowledgement
    */
   @Override
-  protected boolean operateOnPartitionedRegion(ClusterDistributionManager dm, PartitionedRegion r,
-      long startTime) {
+  protected boolean operateOnPartitionedRegion(ClusterDistributionManager dm,
+      PartitionedRegion partitionedRegion, long startTime) {
     if (logger.isTraceEnabled(LogMarker.DM_VERBOSE)) {
       logger.trace(LogMarker.DM_VERBOSE, "ManageBucketMessage operateOnRegion: {}",
-          r.getFullPath());
+          partitionedRegion.getFullPath());
     }
 
-    // This is to ensure that initialization is complete before bucket creation request is
-    // serviced. BUGFIX for 35888
-    if (!r.isInitialized()) {
+    partitionedRegion.checkReadiness(); // Don't allow closed PartitionedRegions that have
```

Comment:
The old code called checkReadiness after inItialized.
The new code calls checkReadiness before isPartitionedRegionReady.
Is this change part of your fix?

Reply:
I think the new order makes more sense